packages/native-stack/src/types.tsx

```diff
@@ -682,6 +708,196 @@ export type NativeStackNavigationOptions = {
+   */
+  width?: number;
+  /**
+   * A boolean value indicating whether the background this item may share with other items in the bar should be hidden.
+   * Only available from iOS 26.0 and later.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uibarbuttonitem/hidessharedbackground
+   */
+  hidesSharedBackground?: boolean;
+  /**
+   * A boolean value indicating whether this bar button item can share a background with other items in a navigation bar or a toolbar.
+   * Only available from iOS 26.0 and later.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uibarbuttonitem/sharesbackground
+   */
+  sharesBackground?: boolean;
+  /**
+   * An identifier used to match bar button items across transitions in a navigation bar or toolbar..
+   * Only available from iOS 26.0 and later.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uibarbuttonitem/identifier
+   */
+  identifier?: string;
+  /**
+   * A badge to be rendered on a bar button item.
+   * Only available from iOS 26.0 and later.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uibarbuttonitembadge
+   */
+  badge?: {
+    /**
+     * The text to display in the badge.
+     */
+    value: string;
+    /**
+     * The color of the badge.
+     */
+    color?: ColorValue;
+    /**
+     * The background color of the badge.
+     */
+    backgroundColor?: ColorValue;
+    style?: {
+      fontFamily?: string;
+      fontSize?: number;
+      fontWeight?: string;
+    };
+  };
+  accessibilityLabel?: string;
+  accessibilityHint?: string;
+}
+
+export interface NativeStackHeaderBarButtonItemWithAction
+  extends SharedHeaderBarButtonItem {
+  /**
+   * Function to call when the button is pressed.
+   */
+  onPress: () => void;
+  /**
+   * A Boolean value that indicates whether the button is in a selected state.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uibarbuttonitem/isselected
+   */
+  selected?: boolean;
+  /**
+   * A Boolean value that indicates whether the button represents an action or selection.
+   * Only available from iOS 15.0 and later.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uibarbuttonitem/changesselectionasprimaryaction
+   */
+  changesSelectionAsPrimaryAction?: boolean;
+}
+
+export interface NativeStackHeaderBarButtonItemMenuAction {
+  title?: string;
+  type: 'action';
+  onPress: () => void;
+  /**
+   * Any SF symbol. Explore them here: https://developer.apple.com/sf-symbols/
+   */
+  sfSymbolName?: string;
+  /**
+   * State of the menu item.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uimenuelement/state
+   */
+  state?: 'on' | 'off' | 'mixed';
+  /**
+   * Style of the menu item.
+   *
+   * Read more: https://developer.apple.com/documentation/uikit/uimenuelement/attributes
+   */
+  attributes?: 'destructive' | 'disabled' | 'hidden' | 'keepsMenuPresented';
```

Comment:
The property name says `attributes`, but it is union, e.g. accepts a single value. Is this intentional?

It's kinda inconsistent that in most places `hidden` is a direct boolean property, and here it is in a property called `attributes`.

Reply:
Thats the way Apple has named it and implemented it. `hidden` is not a boolean but a attribute on UIMenuElement. And as I've said before, my strategy has been not to rename and have similar API as they have created.